### PR TITLE
Fill the section about encoding wraparound check

### PR DESCRIPTION
**Pull request by junyechen1996**

### CODE REVIEW

File: draft-chen-cfrg-vdaf-pine.md [hunk 1]
```diff
@@ -343,38 +358,48 @@ measurement: list[float]) -> list[Field]` that implements this encoding step.
...
-implements this encoding step.
+encoded_gradient: list[Field], wr_joint_rand_xof: Xof) ->
+tuple[list[Field], list[Field]]` that implements this encoding step. It returns
+a tuple of range-checked, wraparound check results, and the wraparound check
+results (i.e., the dot products) from {{run-wr-check}}.
+
+The Client obtains the wraparound check results, as described in
+{{run-wr-check}}. For each wraparound check, the Client runs the range check on
+the result to see if it is in the range of
+`[-wr_check_bound, wr_check_bound + 1]`. Note we choose `wr_check_bound`, such
```
Comment (junyechen1996) [inline draft-chen-cfrg-vdaf-pine.md:390]: I haven't spelled out the derivation of `wr_check_bound` in this PR.

File: draft-chen-cfrg-vdaf-pine.md [hunk 2]
```diff
@@ -343,38 +358,48 @@ measurement: list[float]) -> list[Field]` that implements this encoding step.
...
+Note the Client does not send this dot product to the Aggregators. The
+Aggregators will compute the dot product themselves, based on the encoded
+gradient and the random vector derived on their own.
 
 ### Encoding the Range-Checked, Wraparound Check Results {#encode-wr-check}
 
 We define a function `PineValid.encode_wr_checks(self,
-encoded_gradient: list[Field], wr_joint_rand_xof: Xof) -> list[Field]` that
-implements this encoding step.
+encoded_gradient: list[Field], wr_joint_rand_xof: Xof) ->
```
Comment (junyechen1996) [inline draft-chen-cfrg-vdaf-pine.md:382]: This signature matches our reference implementation.

File: draft-chen-cfrg-vdaf-pine.md [hunk 3]
```diff
@@ -343,38 +358,48 @@ measurement: list[float]) -> list[Field]` that implements this encoding step.
 
 Given the encoded gradient from {{encode-gradient-and-norm}} and the XOF to
 generate the random vectors, the Client needs to run wraparound check
```
Comment (cjpatton) [inline draft-chen-cfrg-vdaf-pine.md:360]:
```suggestion
generate the random vectors, the Client needs to run the wraparound check
```

File: draft-chen-cfrg-vdaf-pine.md [hunk 4]
```diff
@@ -343,38 +358,48 @@ measurement: list[float]) -> list[Field]` that implements this encoding step.
...
-`num_wr_checks` times, and is required to pass at least `num_wr_successes`
-of them. Each wraparound check works as the following:
-
-For each test, the Client generates a random vector with the same dimension as
-the gradient's dimension. Each entry of the random vector is a field element of
-`1` with probability `1/4`, or `0` with probability `1/2`, or `q-1` with
-probability `1/4`, over the field modulus `q`. The Client samples each entry by
-sampling from the XOF output stream two bits at a time:
+`num_wr_checks` times. Each wraparound check works as the following:
```
Comment (cjpatton) [inline draft-chen-cfrg-vdaf-pine.md:361]:
```suggestion
`num_wr_checks` times. Each wraparound check works as follows.
```

File: draft-chen-cfrg-vdaf-pine.md [hunk 5]
```diff
@@ -323,6 +323,21 @@ The measurement encoding is done in two stages:
   Aggregators derive on their own, allow the Aggregators to run wraparound
   checks on their own.
 
+### Encoding Range-Checked Results {#encode-range-check}
```
Comment (cjpatton) [inline draft-chen-cfrg-vdaf-pine.md:326]: This basically says the same thing as the "Range Check" section below. What is your goal exactly?

Comment (junyechen1996) [reply draft-chen-cfrg-vdaf-pine.md:326]: This section is more about Client encoding the range check. While the {{valid-range-check}} talks about what the Aggregator does to verify the range check.

Comment (junyechen1996) [reply draft-chen-cfrg-vdaf-pine.md:326]: I cut some texts in {{valid-range-check}}.

**Review by cjpatton — APPROVED** [review]

**Comment by junyechen1996** [comment]
Squashed.

**Pull request merged**